packages/burner-wallet-connector/src/BurnerWalletConnector.ts

```diff
@@ -0,0 +1,278 @@
+      this.storage.setItem(BURNER_WALLET_ICON, privateKey);
+    }
+
+    if (!privateKey) {
+      this.burnerWallet = Wallet.generate({
+        provider: await this.config.fuelProvider,
+      });
+
+      this.burnerWalletProvider = this.burnerWallet.provider;
+      this.burnerWalletPrivateKey = this.burnerWallet.privateKey;
+
+      this.storage.setItem(
+        BURNER_WALLET_PRIVATE_KEY,
+        this.burnerWalletPrivateKey,
+      );
+
+      return this.burnerWallet;
+    }
+
+    this.burnerWallet = Wallet.fromPrivateKey(
+      privateKey,
+      await this.config.fuelProvider,
+    );
+
+    this.burnerWalletProvider = this.burnerWallet.provider;
+    this.burnerWalletPrivateKey = this.burnerWallet.privateKey;
+
+    return this.burnerWallet;
+  }
+
+  private getStorage(storage?: StorageAbstract) {
+    const _storage =
+      storage ?? (WINDOW.sessionStorage as unknown as StorageAbstract);
+    if (!_storage) {
+      throw new Error('No storage provided');
+    }
+
+    return _storage;
+  }
+
+  /**
+   * ============================================================
+   * Connector methods
+   * ============================================================
+   */
+  async ping(): Promise<boolean> {
+    await this.configFuelProvider();
+    await this.setupBurnerWallet();
+
+    return true;
+  }
+
+  async version(): Promise<Version> {
+    return { app: '0.0.0', network: '0.0.0' };
+  }
+
+  async isConnected(): Promise<boolean> {
+    const connected =
+      (await this.storage.getItem(BURNER_WALLET_CONNECTED)) === 'true';
+
+    if (!connected) {
+      return false;
+    }
+
+    const account = this.burnerWallet?.address.toString();
+
+    return !!account && account.length > 0;
+  }
+
+  async connect(): Promise<boolean> {
+    if (!(await this.isConnected())) {
+      if (!this.burnerWalletProvider) {
+        throw Error('Burner Wallet Provider not found');
+      }
+
+      await this.setupBurnerWallet();
+
+      this.burnerWalletProvider = this.burnerWallet?.connect(
+        this.burnerWalletProvider,
+      ) as Provider;
+
+      this.storage.setItem(BURNER_WALLET_CONNECTED, 'true');
+    }
+
+    this.emit(this.events.connection, true);
+    this.emit(
+      this.events.currentAccount,
+      this.burnerWallet?.address.toAddress(),
+    );
+    this.emit(this.events.accounts, [this.burnerWallet?.address.toAddress()]);
+
+    return this.connected;
+  }
+
+  async accounts(): Promise<string[]> {
+    if (!this.burnerWallet) {
+      throw Error('Wallet not connected');
+    }
+
+    const account = this.burnerWallet.address.toAddress();
+
+    if (!account) {
+      return [];
+    }
+
+    return [account as `fuel${string}`];
+  }
+
+  async disconnect(): Promise<boolean> {
+    if (await this.isConnected()) {
+      this.burnerWalletPrivateKey = null;
+      this.burnerWalletProvider = null;
+      this.burnerWallet = null;
+    }
+
+    this.storage.setItem(BURNER_WALLET_CONNECTED, 'false');
```

Comment:
```suggestion
    this.storage.setItem(BURNER_WALLET_CONNECTED, 'false');
```

The disconnect should also remove the privateKey from the storage.